custom/icds_reports/reports/daily_indicators.py

```diff
@@ -0,0 +1,84 @@
+from custom.icds_reports.models import AggAwcDailyView
+from dateutil.relativedelta import relativedelta
+from io import StringIO
+from custom.icds_reports.utils import apply_exclude
+import csv
+from datetime import date
+
+
+def _get_data_for_daily_indicators(filters, domain, show_test=False):
+    queryset = AggAwcDailyView.objects.filter(
+        **filters
+    ).values('state_name', 'num_launched_awcs', 'daily_attendance_open',
+             'total_eligible_pse', 'total_attended_pse')
+    if not show_test:
+        queryset = apply_exclude(domain, queryset)
+    return queryset
+
+
+def get_daily_indicators(domain):
+    today_date = date.today()
+    filters = {
+        'aggregation_level': 1,
+        'date': today_date
+    }
+    daily_yesterday = None
+    while daily_yesterday is None or (not daily_yesterday and date.day != 1):
+        filename = 'CAS_Daily_Status_{}.csv'.format(filters['date'].strftime('%Y%m%d'))
+        filters['date'] -= relativedelta(days=1)
+        daily_yesterday = _get_data_for_daily_indicators(filters, domain)
```

Comment:
What is this code doing?

Reply:
hey this code is to fetch the data from agg_Awc_daily_view at agg_level=1. It basically start looking from yesterday's date and keep going back in time day by day until it finds the data.

Reply:
Why are we doing that here? If the current numbers are unavailable, we dont want to silently just send yesterday's

Reply:
Its wont be silent. Filename would clearly tell its day before yesterdays data

Reply:
make the api fail when today's data is not present